store/ristretto.go

```diff
@@ -58,15 +61,76 @@ func (s *RistrettoStore) Set(key interface{}, value interface{}, options *Option
 		err = fmt.Errorf("An error has occured while setting value '%v' on key '%v'", value, key)
 	}
 
-	return err
+	if err != nil {
+		return err
+	}
+
+	if tags := options.TagsValue(); len(tags) > 0 {
+		s.setTags(key, tags)
+	}
+
+	return nil
 }
 
-// Delete removes data in Ristretto memoey cache for given key idntifier
+func (s *RistrettoStore) setTags(key interface{}, tags []string) {
+	for _, tag := range tags {
+		var tagKey = fmt.Sprintf(RistrettoTagPattern, tag)
+		var cacheKeys = []string{}
+
+		if result, err := s.Get(tagKey); err == nil {
+			if bytes, ok := result.([]byte); ok {
+				cacheKeys = strings.Split(string(bytes), ",")
+			}
+		}
+
+		var alreadyInserted = false
+		for _, cacheKey := range cacheKeys {
+			if cacheKey == key.(string) {
+				alreadyInserted = true
+				break
+			}
+		}
+
+		if !alreadyInserted {
+			cacheKeys = append(cacheKeys, key.(string))
+		}
+
+		s.Set(tagKey, []byte(strings.Join(cacheKeys, ",")), &Options{
```

Comment:
Storing tags in Ristretto has one potential issue: tags could be evicted from the cache by some reason. In this case tagged objects cannot be invalidated anymore. Im I right or I've missed something?

Reply:
Hi @Magomogo, you are right, using Ristretto in this case you have to ensure your `NumCounters` and `MaxCost` values are properly defined to avoid eviction.

Reply:
Hm..., I still want to have eviction for other data. Maybe you have some ideas how to make certain keys to be "non-evictable"? To me it looks like we shouldn't put the tags information into the same storage at all.

Reply:
Especially considering that "Set" operation in Ristretto is not guaranteed.